### PR TITLE
Add a dedicated service account to run Tekton pipelines

### PR DESCRIPTION
**Pull request by Roming22**
The service account is to be created automatically on tenant namespace creation

E2e tests: https://github.com/codeready-toolchain/toolchain-e2e/pull/718

### CODE REVIEW

**Comment by Roming22** [comment]
@alexeykazakov PR opened: https://github.com/codeready-toolchain/toolchain-e2e/pull/718

File: deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml [hunk 1]
```diff
@@ -269,6 +269,26 @@ objects:
...
+  metadata:
+    name: appstudio-pipelines-runner-rolebinding
+    namespace: ${USERNAME}-tenant
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: appstudio-pipelines-runner-clusterrole
+  subjects:
+    - kind: ServiceAccount
+      apiGroup: rbac.authorization.k8s.io
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml:290]:
```suggestion
      namespace: ${USERNAME}-tenant
```

**Review by alexeykazakov — CHANGES_REQUESTED** [review]
The PR looks good. I'm blocking it just to make sure the PR doesn't get merged until we update the member operator permissions.

File: deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml [hunk 2]
```diff
@@ -269,6 +269,26 @@ objects:
...
+    namespace: ${USERNAME}-tenant
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: appstudio-pipelines-runner-rolebinding
+    namespace: ${USERNAME}-tenant
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: appstudio-pipelines-runner-clusterrole
```
Comment (MatousJobanek) [inline deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml:287]: https://github.com/codeready-toolchain/member-operator/blob/abc7052dba35ef660902664fd843c38a6ec37882/config/appstudio-pipelines-runner/base/appstudio_pipelines_runner_role.yaml#L2-L5
```suggestion
    name: appstudio-pipelines-runner
```

**Comment by MatousJobanek** [comment]
@alexeykazakov for this PR I only updated the name of ClusterRole in the ClusterRoleBinding so it matches what is in the manifests in member-operator 
https://github.com/codeready-toolchain/member-operator/blob/abc7052dba35ef660902664fd843c38a6ec37882/config/appstudio-pipelines-runner/base/appstudio_pipelines_runner_role.yaml#L2-L5
commit: https://github.com/codeready-toolchain/host-operator/pull/797/commits/ed5fce436e0e9029bcc3ed513d83692e38ca98c4

**Review by MatousJobanek — APPROVED** [review]

**Review by alexeykazakov — APPROVED** [review]

**Review by rajivnathan — APPROVED** [review]

**Pull request merged**